Update testing docs (#2098)
pmndrs/zustand: docs/guides/testing.md

The mock provided below will enable the relevant test runner to reset the zustand stores after each test.

> **Warning:** In the following examples we are only supporting the curried version of create `(create()(...))`,

> > due to that is supported on both JavaScript and Typescript. If you are using the uncurried version of

Reviewer: ```suggestion
> due to it being supported by both JavaScript and TypeScript. If you are using the uncurried version of
```

Author: @sewera I changed to `since it is support on` instead of `due to that is`

> to update your hooks to use the curried version or update the mocks in order to support the uncurried
> version.

> 

Reviewer 2: This is looking good. A few tweaks:

> **Warning:** In the following examples we are only demostrating the curried version of create (`create()(...)`), because that is compatible with both 
> JavaScript and Typescript. If you are using the uncurried version of `create(...)` you will need
> to update your hooks to use the curried version, or update the mocks in order to support the uncurried
> version.

Reviewer 2: Thanks for looking through the feedback. 

I think this section "due to that is compatible" is still not quite right? 

Also "both JavaScript and Typescript, If you" is not quite right either. If you want to keep the comma, "If" will need to be "if". However, I think that sentence is very long, and it would be worth breaking it up - similar to the suggestion above. 

What are your thoughts?

Author: @chrisk-7777 what you think about "due to that is supported on both" instead of "due to that is compatible"

Reviewer 2: I think the main issue is "due to that is" doesn't really read well in my opinion. 
I'm happy to a second opinion though :)

Author: @chrisk-7777 because of that doesn't have the same meaning of due to that, does it?

Author: Let me reword it

### Vitest

In the next steps we are going to setup our Vitest environment in order to mock Zustand.

> > **Warning:** In Vitest you can change the [root](https://vitest.dev/config/#root), due to
> > that, you need make sure that you are creating your `__mocks__` directory in the right place.

Reviewer: ```suggestion
> **Warning:** In Vitest you can change the [root](https://vitest.dev/config/#root).
> Due to that, you need make sure that you are creating your `__mocks__` directory in the right place.
```

> that you change the **root** to `./src`, that means you need to create `__mocks__` folder
> under `./src` ending up with `./src/__mocks__` rather than `./__mocks__`. Creating `__mocks__`

> > folder in the wrong place could lead you to some issues while using Vitest.

Reviewer 2: Just a few small tweaks here:

> **Warning:** In Vitest you can change the [root](https://vitest.dev/config/#root), because of this, you
> need make sure that you are creating your `__mocks__` directory in the right location. Let's say
> that you change the **root** to `./src`, that means you need to create a `__mocks__` directory
> under `./src`. The end result would be `./src/__mocks__`, rather than `./__mocks__`. Creating a `__mocks__`
> directory in the wrong location could lead to issues while using Vitest.

> > Creating `__mocks__` directory in the wrong place could lead to issues while using Vitest.

Reviewer: ```suggestion
> Creating `__mocks__` directory in the wrong place can lead to issues when using Vitest.
```
